crates/sparrow-runtime/src/prepare/preparer.rs

```diff
@@ -261,19 +268,51 @@ fn get_required_column<'a>(
         .ok_or_else(|| error_stack::report!(Error::BatchMissingRequiredColumn(name.to_owned())))
 }
 
-fn time_multiplier(time_unit: Option<&str>) -> error_stack::Result<Option<i64>, Error> {
-    match time_unit.unwrap_or("ns") {
-        "ns" => Ok(None),
-        "us" => Ok(Some(1_000)),
-        "ms" => Ok(Some(1_000_000)),
-        "s" => Ok(Some(1_000_000_000)),
+fn time_multiplier(
```

Comment:
This will conflict with #786 , but I think it's blocking Eric so I want to merge it in sooner